### PR TITLE
Add the gh-pages Workflow

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

File: .github/workflows/gh-pages.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+on:
+  push:
+    branches:
+    - 'master'
+
+jobs:
+  docs:
+    name: "Deploy"
+    runs-on: ubuntu-latest
```
Comment (traversaro) [inline .github/workflows/gh-pages.yml:11]: For reproducibility, I suggest to use `ubuntu-18.04`  or `ubuntu-20.04` .

File: .github/workflows/gh-pages.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    steps:
+      - uses: actions/checkout@master
+      - uses: webfactory/ssh-agent@v0.3.0
+        with:
+          ssh-private-key: ${{ secrets.SSH_PRIVATE_GH_PAGES }}
+      - name: Dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y xsltproc doxygen doxygen-doc texlive ghostscript graphviz
```
Comment (traversaro) [inline .github/workflows/gh-pages.yml:21]: Are all of these necessary?

Comment (GiulioRomualdi) [reply .github/workflows/gh-pages.yml:21]: I think we only need `doxygen` and `graphviz`. Right now we are also generating the documentation in `latex` (i.e. check [here](https://github.com/robotology/osqp-eigen/tree/gh-pages/doxygen/doc/latex)). I think we can drop it.

File: .github/workflows/gh-pages.yml (hunk 1 above)
Comment (GiulioRomualdi) [reply .github/workflows/gh-pages.yml]: Done in https://github.com/robotology/osqp-eigen/pull/65/commits/1e3ed3badd2bebcf961e7a46624d0a7a69bef0bd

**Pull request merged**